syntax = "proto3";

package calcs;

service Calcs {
    rpc GetBhaskara (BhaskaraRequest) returns (RootsReply) {}
    rpc GetFibonacci (FibonacciRequest) returns (stream FibonacciReply) {}
}

message BhaskaraRequest {
    int32 var_a = 1;
    int32 var_b = 2;
    int32 var_c = 3;
}

message RootsReply {
    float x1 = 1;
    float x2 = 2;
}
  
message FibonacciRequest {
    int32 limit = 1;
    int32 delay = 2;
}

message FibonacciReply {
    int32 value = 1;
}